topics/authentication.md

```diff
@@ -6,140 +6,151 @@ Code examples: <a href="samples.md" anchor="authentication">Authentication</a>
+
+### Session {id="sessions"}
+[Sessions](sessions.md) provide a mechanism to persist data between different HTTP requests. Typical use cases include storing a logged-in user's ID, the contents of a shopping basket, or keeping user preferences on the client. In Ktor, a user that already has an associated session can be authenticated using the `session` provider. Learn how to do this from [](session-auth.md).
 
-It can be used in some cases along with the [sessions plugin](sessions.md)
-to keep the login information between requests.
 
 ## Add dependencies {id="add_dependencies"}
 To enable authentication, you need to include the `ktor-auth` artifact in the build script:
 
 <var name="artifact_name" value="ktor-auth"/>
 <include src="lib.xml" include-id="add_ktor_artifact"/>
 
-Note that some authentication plugins (for example [JWT/JWK](jwt.md) and [LDAP](ldap.md)) might require additional artifacts.
+Note that some authentication providers, such as [JWT](jwt.md) and [LDAP](ldap.md), require additional artifacts.
+
 
 
-## Basic usage
+## Install Authentication {id="install"}
+<var name="feature_name" value="Authentication"/>
+<include src="lib.xml" include-id="install_feature"/>
 
-Ktor defines two concepts: credentials and principals.
 
-* A principal is something that can be authenticated: a user, a computer, a group, etc.
-* A credential is an object that represents a set of properties for the server to authenticate a principal:
-  a user/password, an API key or an authenticated payload signature, etc.
+## Configure Authentication {id="configure"}
+After [installing Authentication](#install), you can configure and use `Authentication` as follows:
 
-To install it, you have to call to `application.install(Authentication)`. You have to install this plugin
-directly to the application and it *won't* work in another `ApplicationCallPipeline` like `Route`.
 
->You might still be able to call the install code inside a Route if you have the Application injected in a nested DSL,
->but it will be applied to the application itself.
->
-{type="note"}
+### Step 1: Choose an authentication provider {id="choose-provider"}
 
-Using its DSL, it allows you to configure the authentication providers available:
+To use a specific authentication provider ([basic](basic.md), [digest](digest.md), [form](form.md), and so on), you need to call a corresponding function inside the `install` block. For example, to use the `basic` authentication, call the [basic](https://api.ktor.io/ktor-features/ktor-auth/ktor-auth/io.ktor.auth/basic.html) function:
 
 ```kotlin
+install(Authentication) {
+    basic {
+        // [[[Configure basic authentication|basic.md]]]
+    }
+}
 ```
-{src="snippets/auth-basic/src/main/kotlin/com/example/Application.kt" lines="9-18"}
+Inside this function, you can [configure](#configure-provider) settings specific to this provider.
+
 
+### Step 2: Specify a provider name {id="provider-name"}
```

Comment:
I'm not sure if this should be a step. It's somewhat confusing that it's a step that is optional and actually related to previous step. I'd just make it part of the first Step and start with "If you have multiple authentication providers, whether they are of the same type or not, it is good practice to name them so that you can refer to them later in the code by name", and then proceed with the sample.